### PR TITLE
Update readme based on the internal redesign

### PR DESCRIPTION
**Pull request by frederikprijck**
### Description

Restructures the readme file based on our internal design to try and create a uniform experience across the readme's of our different SDKs.

- [x] Add header image

### Checklist

- [x] I have added documentation for new/changed functionality in this PR or in auth0.com/docs
- [x] All active GitHub checks for tests, formatting, and security are passing
- [x] The correct base branch is being used, if not the default branch

### CODE REVIEW

**Comment by stevehobbsdev** [comment]
Looks good. The API reference methods aren't explicitly called out like they are in the Figma example, are we leaving this out?

> See the Figma for an example of this. Again the NextJS repo provides a good example. Each endpoint reference is essentially a link that takes the developer to the API reference page.

**Comment by frederikprijck** [comment]
I added a small section for the API reference. The AuthService and AuthConfig are the main points that require elaboration on the API. The Guard and Interceptor follow Angular patterns and are not something the user interacts with.

File: README.md [hunk 1]
```diff
@@ -1,36 +1,30 @@
...
-[![CircleCI](https://img.shields.io/circleci/build/github/auth0/auth0-angular)](https://circleci.com/gh/auth0/auth0-angular)
-[![Codecov](https://img.shields.io/codecov/c/github/auth0/auth0-angular)](https://codecov.io/gh/auth0/auth0-angular)
-![Downloads](https://img.shields.io/npm/dw/@auth0/auth0-angular)
-
 # Auth0 Angular SDK
 
 A library for integrating [Auth0](https://auth0.com) into an Angular application.
 
-## Table of Contents
+![Release](https://img.shields.io/github/v/release/auth0/auth0-angular)
```
Comment (ewanharris) [inline README.md]: One question here, do we want `Release` to show the the latest release from a time perspective (i.e. including betas, previous majors that have been updated) or from a "latest GA release" perspective? I think using GitHub for the source gives us the former whereas npm/other package managers would give the latter

 For example [spa-js](https://github.com/auth0/auth0-spa-js/tree/master) currently shows `2.0.0-beta.1` which while technically correct doesn't feel like the version we want shown (1.22.5 imo)

Comment (frederikprijck) [reply README.md]: That's a good point. Let me update any of the open PR's I have that use this, thanks!

**Review by ewanharris — APPROVED** [review]

**Pull request merged**